### PR TITLE
Add lmi-dist integration tests

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: tests/integration/llm/prepare.py [hunk 1]
```diff
@@ -350,6 +350,34 @@
     }
 }
 
+lmi_dist_model_list = {
+    "gpt-neox-20b": {
+        "option.model_id": "EleutherAI/gpt-neox-20b",
+        "option.tensor_parallel_degree": 4,
+        "option.max_rolling_batch_size": 4,
+        "load_on_devices": 0
```
Comment (sindhuvahinis) [inline tests/integration/llm/prepare.py:358]: we dont need to add load_on_devices now. We can also remove the entryPoint.

File: .github/workflows/lmi_dist_integration.yml [hunk 2]
```diff
@@ -2,6 +2,11 @@ name: LMI Dist library Unit/Integration tests
 
 on:
   workflow_dispatch:
+    inputs:
```
Comment (lanking520) [inline .github/workflows/lmi_dist_integration.yml:5]: since we anyway run the test with the full model server, should we just merge with @sindhuvahinis 's pipeline?

File: tests/integration/llm/prepare.py (hunk 1 above)
Comment (xyang16) [reply tests/integration/llm/prepare.py:358]: Changed.

File: .github/workflows/lmi_dist_integration.yml (hunk 2 above)
Comment (xyang16) [reply .github/workflows/lmi_dist_integration.yml]: Done

File: .github/workflows/rolling_batch_integration.yml [hunk 3]
```diff
@@ -159,6 +159,77 @@ jobs:
...
+        if: ${{ failure() }}
+        working-directory: tests/integration
+        run: |
+          docker rm -f $(docker ps -aq) || true
+          cat logs/serving.log
+      - name: Upload test logs
+        uses: actions/upload-artifact@v3
+        with:
+          name: lmi-dist-logs
+          path: tests/integration/logs/
```
Comment (lanking520) [inline .github/workflows/rolling_batch_integration.yml:232]: need to add your test to `stop-runners` list

Comment (xyang16) [reply .github/workflows/rolling_batch_integration.yml:232]: Done.

**Review by lanking520 — APPROVED** [review]
